brayns/Brayns.cpp

```diff
@@ -426,159 +369,114 @@ struct Brayns::Impl : public PluginAPI
-            _loadNESTCircuit();
-            loadingProgress += tic;
-        }
-
-        if (!geometryParameters.getCircuitConfiguration().empty())
-            _loadCircuitConfiguration(updateProgress);
+        registry.registerLoader(
+            {MorphologyLoader::getName(),
+             MorphologyLoader::getSupportedDataTypes(), [&] {
+                 return std::make_unique<MorphologyLoader>(scene,
+                                                           geometryParameters);
+             }});
+
+        registry.registerLoader(
+            {CircuitLoader::getName(), CircuitLoader::getSupportedDataTypes(),
+             [&] {
+                 return std::make_unique<CircuitLoader>(
+                     scene, _parametersManager.getApplicationParameters(),
+                     geometryParameters);
+             }});
 #endif
 
-        if (!geometryParameters.getMolecularSystemConfig().empty())
-            _loadMolecularSystem(updateProgress);
-
-        scene.saveToCacheFile();
-        scene.buildEnvironmentMap();
-        scene.markModified();
+        registry.registerLoader(
+            {MolecularSystemReader::getName(),
+             MolecularSystemReader::getSupportedDataTypes(), [&] {
+                 return std::make_unique<MolecularSystemReader>(
+                     scene, geometryParameters);
+             }});
     }
 
-#if (BRAYNS_USE_BRION)
-    /**
-     * Loads data from a NEST circuit file (command line parameter
-     * --nest-circuit)
-     */
-    void _loadNESTCircuit()
+    void _loadData()
     {
         auto& geometryParameters = _parametersManager.getGeometryParameters();
+        auto& sceneParameters = _parametersManager.getSceneParameters();
         auto& scene = _engine->getScene();
 
-        const std::string& circuit(geometryParameters.getNESTCircuit());
-        if (!circuit.empty())
+        const auto& paths =
+            _parametersManager.getApplicationParameters().getInputPaths();
+        if (!paths.empty())
         {
-            NESTLoader loader(scene, geometryParameters);
+            if (paths.size() == 1 && paths[0] == "demo")
+            {
+                _engine->getScene().buildDefault();
+                _engine->getScene().buildEnvironmentMap();
+                return;
+            }
 
-            // need to import circuit first to determine _frameSize for report
-            // loading
-            auto model = loader.importFromFile(circuit);
+            for (const auto& path : paths)
+            {
+                AddModelTask task({path}, _engine);
+                task.result();
+            }
+        }
 
-            const std::string& cacheFile(geometryParameters.getNESTCacheFile());
-            if (!geometryParameters.getNESTReport().empty() &&
-                cacheFile.empty())
-                throw std::runtime_error(
-                    "Need cache file to visualize simulation data");
+        if (!geometryParameters.getCircuitConfiguration().empty())
```

Comment:
Let's remove this and the `getCircuitConfiguration()`; circuits can already be loaded by regular `--input-paths`